### PR TITLE
DOC-105: Pointers JSDoc revision

### PR DESCRIPTION
**Pull request by ctrlaltdavid**
Namepaces:
- Pointers
- LaserPointers

Other:
- Picks API items updated to match changes in Pointers API.
- RayPick API items updated to match changes in Pointers API.

Case: https://highfidelity.atlassian.net/browse/DOC-105

Note: No executable code changes.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20207/HighFidelity-Beta-PR15865-5c29eb0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20207/HighFidelity-Beta-Interface-PR15865-5c29eb0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20207/HighFidelity-Beta-PR15865-5c29eb0.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20207/HighFidelity-Beta-Interface-PR15865-5c29eb0.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20207/HighFidelity-Beta-PR15865-5c29eb0.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20207/HighFidelity-Quest-Beta-PR15865-5c29eb0.apk)

File: interface/src/raypick/PointerScriptingInterface.cpp [hunk 1]
```diff
@@ -104,48 +105,76 @@ unsigned int PointerScriptingInterface::createStylus(const QVariant& properties)
...
- * When returned from {@link Pointers.getPointerProperties}, a map between render state names and DefaultRayPointRenderStates.
- * @property {boolean} [hover=false] If this Pointer should generate hover events.
- * @property {Pointers.Trigger[]} [triggers] A list of different triggers mechanisms that control this Pointer's click event generation.
+ * The properties of a ray pointer. These include the properties from the underlying ray pick that the pointer uses.
+ * @typedef {object} Pointers.RayPointerProperties
+ * @property {boolean} [faceAvatar=false] - <code>true</code> if the overlay rendered at the end of the ray rotates about the 
+ *     world y-axis to always face the avatar; <code>false</code> if it maintains its world orientation.
+ * @property {boolean} [centerEndY=true] - <code>true</code> if the overlay rendered at the end of the ray is centered on 
+ *     the ray end; <code>false</code> if the overlay is rendered adjacent to the surface if <code>followNormal</code> is 
+ *     <code>true</code> or on top of the ray end if if <code>followNormal</code> is <code>false</code>.
```
Comment (jennaingersoll) [inline interface/src/raypick/PointerScriptingInterface.cpp]: Not sure what this is saying:

<code>false</code> if the overlay is rendered adjacent to the surface if <code>followNormal</code> is 
<code>true</code> or on top of the ray end if if <code>followNormal</code> is <code>false</code>.

File: interface/src/raypick/PointerScriptingInterface.cpp [hunk 2]
```diff
@@ -260,58 +289,84 @@ unsigned int PointerScriptingInterface::createLaserPointer(const QVariant& prope
...
-* When returned from {@link Pointers.getPointerProperties}, a map between render state names and DefaultParabolaPointerRenderStates.
-* @property {boolean} [hover=false] If this Pointer should generate hover events.
-* @property {Pointers.Trigger[]} [triggers] A list of different triggers mechanisms that control this Pointer's click event generation.
-*/
+ * The properties of a parabola pointer. These include the properties from the underlying parabola pick that the pointer uses.
+ * @typedef {object} Pointers.ParabolaPointerProperties
+ * @property {boolean} [faceAvatar=false] - <code>true</code> if the overlay rendered at the end of the ray rotates about the
+ *     world y-axis to always face the avatar; <code>false</code> if it maintains its world orientation.
+ * @property {boolean} [centerEndY=true] - <code>true</code> if the overlay rendered at the end of the ray is centered on
+ *     the ray end; <code>false</code> if the overlay is rendered adjacent to the surface if <code>followNormal</code> is
```
Comment (jennaingersoll) [inline interface/src/raypick/PointerScriptingInterface.cpp]: Same comment as above.

File: interface/src/raypick/PointerScriptingInterface.h [hunk 3]
```diff
@@ -35,184 +36,416 @@ class PointerScriptingInterface : public QObject, public Dependency {
...
-    * @property {Controller.Standard|Controller.Actions|function} action This can be a built-in Controller action, like Controller.Standard.LTClick, or a function that evaluates to >= 1.0 when you want to trigger <code>button</code>.
-    * @property {string} button Which button to trigger.  "Primary", "Secondary", "Tertiary", and "Focus" are currently supported.  Only "Primary" will trigger clicks on web surfaces.  If "Focus" is triggered,
-    * it will try to set the entity focus to the object at which the Pointer is aimed.  Buttons besides the first three will still trigger events, but event.button will be "None".
-    */
+     * Specifies that a {@link Controller} action or function should trigger events on the entity or overlay currently 
+     * intersected by a {@link Pointers.RayPointerProperties|Ray} or {@link Pointers.ParabolaPointerProperties|Parabola} 
+     * pointer.
+     * @typedef {object} Pointers.Trigger
+     * @property {Controller.Standard|Controller.Actions|function} action - The controller output or function that triggers the 
+     *     events on the entity or overlay. If a function, it must return a number <code>>= 1.0</code> to start the action and
```
Comment (jennaingersoll) [inline interface/src/raypick/PointerScriptingInterface.h]:
```suggestion
     *     events on the entity or overlay. If a function, it must return a number <code>&gt;= 1.0</code> to start the action and 
```

File: interface/src/raypick/PointerScriptingInterface.h [hunk 4]
```diff
@@ -35,184 +36,416 @@ class PointerScriptingInterface : public QObject, public Dependency {
...
-    * @property {string} button Which button to trigger.  "Primary", "Secondary", "Tertiary", and "Focus" are currently supported.  Only "Primary" will trigger clicks on web surfaces.  If "Focus" is triggered,
-    * it will try to set the entity focus to the object at which the Pointer is aimed.  Buttons besides the first three will still trigger events, but event.button will be "None".
-    */
+     * Specifies that a {@link Controller} action or function should trigger events on the entity or overlay currently 
+     * intersected by a {@link Pointers.RayPointerProperties|Ray} or {@link Pointers.ParabolaPointerProperties|Parabola} 
+     * pointer.
+     * @typedef {object} Pointers.Trigger
+     * @property {Controller.Standard|Controller.Actions|function} action - The controller output or function that triggers the 
+     *     events on the entity or overlay. If a function, it must return a number <code>>= 1.0</code> to start the action and 
+     *     <code>< 1.0</code> to terminate the action.
```
Comment (jennaingersoll) [inline interface/src/raypick/PointerScriptingInterface.h]:
```suggestion
     *     <code>&lt;1.0</code> to terminate the action.
```

File: interface/src/raypick/PointerScriptingInterface.h (hunk 3 above)
Comment (ctrlaltdavid) [reply interface/src/raypick/PointerScriptingInterface.h]: Done.

File: interface/src/raypick/PointerScriptingInterface.h (hunk 4 above)
Comment (ctrlaltdavid) [reply interface/src/raypick/PointerScriptingInterface.h]: Done.

File: interface/src/raypick/PointerScriptingInterface.cpp (hunk 1 above)
Comment (ctrlaltdavid) [reply interface/src/raypick/PointerScriptingInterface.cpp]: Reworded.

File: interface/src/raypick/PointerScriptingInterface.cpp (hunk 2 above)
Comment (ctrlaltdavid) [reply interface/src/raypick/PointerScriptingInterface.cpp]: Reworded.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20251/HighFidelity-Beta-PR15865-8fb83e9.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20251/HighFidelity-Beta-Interface-PR15865-8fb83e9.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20251/HighFidelity-Beta-PR15865-8fb83e9.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20251/HighFidelity-Beta-Interface-PR15865-8fb83e9.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20251/HighFidelity-Beta-PR15865-8fb83e9.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20251/HighFidelity-Quest-Beta-PR15865-8fb83e9.apk)

**Pull request merged**